compiler/optimizer/OMROptimizer.hpp

```diff
@@ -217,6 +221,9 @@ class Optimizer
    bool canRunBlockByBlockOptimizations()          { return _canRunBlockByBlockOptimizations; }
    void setCanRunBlockByBlockOptimizations(bool v) { _canRunBlockByBlockOptimizations = v; }
 
+   bool callsAsUsesEnabled() { return _enableCallsAsUses; }
+   void setEnableCallsAsUses(bool v) { _enableCallsAsUses = v; }
```

Comment:
The problem I have with this API is that the whether calls should be uses is a one time thing. That means we wither need to set in the constructor or we need to set it in the `createUseDefInfo` method and allow optimizers to replace that method. I think this style of get/set is dangers because it would mean we could allow this to change during compilation which would be dangerous. I'd like to see these removed since there should be no need to interrospect this or update this outside of the optimizer itself.

Reply:
Hi @andrewcraik, I was wondering whether to add this to the constructor or let derived classes override the `createUseDefInfo` method. I wasn't sure as I noticed many other flags are present and it seemed like existing practice to use an api to update the flag. In the end because of this I added this api so that the language front end can update the flag. But you are right that this is a one time thing. To allow derived classes to override the method I would need to make the method virtual.  Or else make it a constructor argument that takes false as the default value.

Reply:
Another option would be to have a virtual method that returns a constant false called `getCallsAsUses` which defaults to false and which deriving implementations could override with a comment that it should return a constant true or false would be one way around it. Since it is a constant per project I don't think it needs the storage or the API to get/set since it would be problematic to allow it to vary. I guess I am pushing for an API design that forces correctness by construction if that makes sense.

Reply:
I have made it a virtual function as suggested.